test/WebJobs.Script.Tests.Integration/Host/SecretsRepositoryTests.cs

```diff
@@ -370,6 +369,133 @@ public async Task GetSecretSnapshots_ReturnsExpected(SecretsRepositoryType repos
             }
         }
 
+        [Theory]
+        [InlineData(SecretsRepositoryType.BlobStorage)]
+        [InlineData(SecretsRepositoryType.BlobStorageSas)]
+        public async Task BlobRepository_WriteAsync_DoesNot_ClearBlobContents(SecretsRepositoryType repositoryType)
+        {
+            using (var directory = new TempDirectory())
+            {
+                await _fixture.TestInitialize(repositoryType, directory.Path);
+
+                ScriptSecrets testSecrets = new HostSecrets()
+                {
+                    MasterKey = new Key("master", "test"),
+                    FunctionKeys = new List<Key>() { new Key(KeyName, "test") },
+                    SystemKeys = new List<Key>() { new Key(KeyName, "test") }
+                };
+
+                string testFunctionName = "host";
+
+                var target = _fixture.GetNewSecretRepository();
+
+                // Set up initial secrets.
+                await _fixture.WriteSecret(testFunctionName, testSecrets);
+
+                // Perform a write and read similtaneously. Previously, our usage of OpenWriteAsync
+                // would erase the content of the blob while writing, resulting in null secrets from the
+                // read.
+                Task writeTask = target.WriteAsync(ScriptSecretsType.Host, testFunctionName, testSecrets);
+                HostSecrets secretsContent = await target.ReadAsync(ScriptSecretsType.Host, testFunctionName) as HostSecrets;
+
+                await writeTask;
+
+                Assert.Equal(secretsContent.MasterKey.Name, "master");
+                Assert.Equal(secretsContent.MasterKey.Value, "test");
+                Assert.Equal(secretsContent.FunctionKeys[0].Name, KeyName);
+                Assert.Equal(secretsContent.FunctionKeys[0].Value, "test");
+                Assert.Equal(secretsContent.SystemKeys[0].Name, KeyName);
+                Assert.Equal(secretsContent.SystemKeys[0].Value, "test");
+            }
+        }
+
+        [Theory]
+        [InlineData(SecretsRepositoryType.BlobStorage)]
+        [InlineData(SecretsRepositoryType.BlobStorageSas)]
+        public async Task BlobRepository_SimultaneousWrites_Throws_PreconditionFailed(SecretsRepositoryType repositoryType)
```

Comment:
With the previous code in `WriteToBlobAsync()`, this test behaved the same. I wanted to ensure no change in behavior.